pallets/handles/src/lib.rs

```diff
@@ -295,14 +297,17 @@ pub mod pallet {
 		fn generate_suffix_for_canonical_handle(
 			canonical_handle: &str,
 			cursor: usize,
-		) -> HandleSuffix {
+		) -> Result<u16, DispatchError> {
 			let mut suffix_generator = SuffixGenerator::new(
 				T::HandleSuffixMin::get() as usize,
 				T::HandleSuffixMax::get() as usize,
 				&canonical_handle,
 			);
-			let sequence: Vec<usize> = suffix_generator.suffix_iter().collect();
-			sequence[cursor] as HandleSuffix
+			let suffix_option = suffix_generator.suffix_iter().nth(cursor);
+			match suffix_option {
+				Some(suffix) => Ok(suffix as u16),
+				None => Err(Error::<T>::SuffixesExhausted.into()),
+			}
```

Comment:
Better safely check again than simple unwrap and also go to the the nth index directly over collecting everything